service/storage_service.cc

```diff
@@ -7193,7 +7193,20 @@ future<join_node_response_result> storage_service::join_node_response_handler(jo
             slogger.info("raft topology: coordinator accepted request to join, "
                     "waiting for nodes {} to be alive before responding and continuing",
                     sync_nodes);
-            co_await _gossiper.wait_alive(sync_nodes, wait_for_live_nodes_timeout);
+            try {
+                co_await _gossiper.wait_alive(sync_nodes, wait_for_live_nodes_timeout);
+            } catch (std::runtime_error& e) {
+                const sstring message = ::format(
+                        "waiting for nodes {} to be alive failed after coordinator accepted request to join: {}. "
+                        "Make sure all nodes are alive before restarting this node.", sync_nodes, e.what());
+                slogger.warn("raft topology: {}", message);
+                e = std::runtime_error(message);
+
+                _join_node_response_done.set_exception(std::make_exception_ptr(e));
+
+                throw e;
```

Comment:
The idea was to throw a new exception with more detailed message. I assigned it on the caught exception, but you explained why it is a bad idea in the previous comments. Of course, I didn't have to do it. I could simply create a new variable.

Anyway, since the exception handler should cover the entirety of the `overloaded_functor` in the `join_node_response_handler`, I can't create a more detailed message that makes sense only if `_gossiper.wait_alive` throws an exception. I will just rethrow the caught exception.